src/Fusonic/Linq/Linq.php

```diff
@@ -135,7 +136,8 @@ public function take($count)
             return new Linq([]);
         }
 
-        return new Linq(new \LimitIterator($this->iterator, 0, $count));
+        // IteratorIterator wraps $this->iterator because it might be Traversable but not an Iterator.
```

Comment:
Please include this url in the comment "https://bugs.php.net/bug.php?id=52280" so that there is more context.